scripts/linters/pre_commit_linter.py

```diff
@@ -0,0 +1,517 @@
+        '.py': [], '.html': [], '.ts': [], '.js': [], 'other': [], '.css': []
+    }
+    for file_path in file_paths:
+        _, extension = os.path.splitext(file_path)
+        if extension in all_filepaths_dict:
+            all_filepaths_dict[extension].append(file_path)
+        else:
+            all_filepaths_dict['other'].append(file_path)
+    _FILES.update(all_filepaths_dict)
+
+
+def _print_complete_summary_of_errors(all_messages):
+    """Print complete summary of errors."""
+    error_messages = all_messages
+    if error_messages != '':
+        python_utils.PRINT('Summary of Errors:')
+        python_utils.PRINT('----------------------------------------')
+        for message in error_messages:
+            python_utils.PRINT(message)
+
+
+def main(args=None):
+    """Main method for pre commit linter script that lints Python, JavaScript,
+    HTML, and CSS files.
+    """
+    parsed_args = _PARSER.parse_args(args=args)
+    # File extension to be linted.
+    file_extension_types = _get_file_extensions(
+        parsed_args.only_check_file_extensions)
+    # Default mode is non-verbose mode, if arguments contains --verbose flag it
+    # will be made True, which will represent verbose mode.
+    verbose_mode_enabled = bool(parsed_args.verbose)
+    all_filepaths = _get_all_filepaths(parsed_args.path, parsed_args.files)
+
+    python_utils.PRINT('Starting Linter....')
+
+    if len(all_filepaths) == 0:
+        python_utils.PRINT('---------------------------')
+        python_utils.PRINT('No files to check.')
+        python_utils.PRINT('---------------------------')
+        return
+
+    read_files(all_filepaths)
+    categorize_files(all_filepaths)
+
+    third_party_max_concurrent_runs = 2
+
+    # Prepare tasks.
+    concurrent_count = min(
+        multiprocessing.cpu_count(), third_party_max_concurrent_runs)
+    semaphore = threading.Semaphore(concurrent_count)
+
+    custom_linters = []
+    third_party_linters = []
+    for file_extension_type in file_extension_types:
+        custom_linter, third_party_linter = _lint_all_files(
+            _FILES['.js'], _FILES['.ts'], _FILES['.py'], _FILES['.html'],
+            _FILES['.css'], file_extension_type,
+            verbose_mode_enabled=verbose_mode_enabled)
+        custom_linters += custom_linter
+        third_party_linters += third_party_linter
+
+    tasks_custom = []
+    tasks_third_party = []
+
+    for linter in custom_linters:
+        task_custom = concurrent_task_utils.create_task(
+            linter.perform_all_lint_checks,
+            verbose=verbose_mode_enabled, name='custom')
+        tasks_custom.append(task_custom)
+
+    for linter in third_party_linters:
+        task_third_party = concurrent_task_utils.create_task(
+            linter.perform_all_lint_checks,
+            verbose=verbose_mode_enabled, semaphore=semaphore,
+            name='third_party')
+        tasks_third_party.append(task_third_party)
+
+    # Concurrency limit: 25.
```

Comment:
Where are these concurrency limits defined? I do not see it in the arguments; it's hard to tell from the way this is written that the "semaphore" in the second line means "2".

I think you should standardize the way these are called. Make semaphore a mandatory argument for both, for example, or (maybe better) pass in the desired concurrency limit as a number and let concurrent_task_utils handle the semaphore.

Reply:
Done.